### PR TITLE
HBASE-23380: General Cleanup of FSUtil

### PR DESCRIPTION
**Pull request by belugabehr**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  1s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 40s |  master passed  |
| +1 :green_heart: |  compile  |   0m 59s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 56s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 40s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 36s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 34s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 35s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 59s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 24s |  hbase-server: The patch generated 3 new + 66 unchanged - 8 fixed = 69 total (was 74)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 56s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 54s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 20s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 147m 43s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 207m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/912 |
| JIRA Issue | HBASE-23380 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 048aa45c03d4 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-912/out/precommit/personality/provided.sh |
| git revision | master / 9c82a65b29 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/1/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/1/testReport/ |
| Max. process+thread count | 4956 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 55s |  master passed  |
| +1 :green_heart: |  compile  |   1m  4s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 42s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 38s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 44s |  master passed  |
| +0 :ok: |  spotbugs  |   5m  7s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m  5s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 31s |  hbase-server: The patch generated 0 new + 66 unchanged - 8 fixed = 66 total (was 74)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 40s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  19m 40s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 42s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 10s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 147m 31s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 34s |  The patch does not generate ASF License warnings.  |
|  |   | 217m 49s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.io.hfile.bucket.TestBucketCache |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/912 |
| JIRA Issue | HBASE-23380 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4426cf392a7c 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-912/out/precommit/personality/provided.sh |
| git revision | master / 8548463e74 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/2/testReport/ |
| Max. process+thread count | 4355 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java [hunk 1]
```diff
@@ -227,12 +232,8 @@ public static FSDataOutputStream create(Configuration conf, FileSystem fs, Path
           throw new IOException(ite.getCause());
         } catch (NoSuchMethodException e) {
           LOG.debug("DFS Client does not support most favored nodes create; using default create");
-          if (LOG.isTraceEnabled()) LOG.trace("Ignoring; use default create", e);
-        } catch (IllegalArgumentException e) {
-          LOG.debug("Ignoring (most likely Reflection related exception) " + e);
-        } catch (SecurityException e) {
-          LOG.debug("Ignoring (most likely Reflection related exception) " + e);
-        } catch (IllegalAccessException e) {
+          LOG.trace("Ignoring; use default create", e);
```
Comment (HorizonNet) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: Why did you remove the check if trace is enabled?

Comment (belugabehr) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: For such a trivial case, there is not need to check for trace being enabled.  The trace method itself will check internally if trace is enabled.  Checking for trace logging here would only be required if there was some sort of complicated computation that would simply be thrown away every time.  Passing a static string and a reference to an Exception does not require any computation.

**Comment by belugabehr** [comment]
@HorizonNet Thanks for the review! Much appreciated.

**Comment by belugabehr** [comment]
Rebased on master

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  6s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 52s |  master passed  |
| +1 :green_heart: |  compile  |   0m 56s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 23s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 44s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 41s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 16s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 14s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  hbase-server: The patch generated 0 new + 66 unchanged - 8 fixed = 66 total (was 74)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 43s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m  9s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 156m 50s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 216m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/912 |
| JIRA Issue | HBASE-23380 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7a71b0a72cae 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-912/out/precommit/personality/provided.sh |
| git revision | master / 82e155eb26 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/3/testReport/ |
| Max. process+thread count | 4466 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java [hunk 2]
```diff
@@ -138,20 +137,26 @@ public static boolean isDistributedFileSystem(final FileSystem fs) throws IOExce
...
     String toSearchName;
     boolean result = false;
+
+    if (pathToSearch.depth() != pathTail.depth()) {
+      return false;
+    }
+
     do {
       tailName = tailPath.getName();
-      if (tailName == null || tailName.length() <= 0) {
```
Comment (busbey) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: Does the `Path` API promise that `getName` won't return null for the versions of Hadoop in our support matrix?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java [hunk 3]
```diff
@@ -546,19 +544,19 @@ public static void setVersion(FileSystem fs, Path rootdir, String version,
...
     while (true) {
       try {
         Path filePath = new Path(rootdir, HConstants.CLUSTER_ID_FILE_NAME);
         return fs.exists(filePath);
       } catch (IOException ioe) {
-        if (wait > 0) {
-          LOG.warn("Unable to check cluster ID file in " + rootdir.toString() +
-              ", retrying in "+wait+"msec: "+StringUtils.stringifyException(ioe));
+        if (wait > 0L) {
+          LOG.warn("Unable to check cluster ID file in {}, retrying in {}ms", ioe);
```
Comment (busbey) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: this is missing the params for `rootdir` and `wait`.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java [hunk 4]
```diff
@@ -771,7 +770,7 @@ static public void addToHDFSBlocksDistribution(
   public static int getTotalTableFragmentation(final HMaster master)
   throws IOException {
     Map<String, Integer> map = getTableFragmentation(master);
-    return map != null && map.size() > 0 ? map.get("-TOTAL-") : -1;
```
Comment (busbey) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: please update the javadocs for `getTableFragmentation` to say it can't return `null` (AFAICT it doesn't currently).

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java [hunk 5]
```diff
@@ -940,7 +939,7 @@ protected boolean isValidName(final String name) {
       try {
         TableName.isLegalTableQualifierName(Bytes.toBytes(name));
       } catch (IllegalArgumentException e) {
-        LOG.info("INVALID NAME " + name);
+        LOG.info("Invalid table name: " + name);
```
Comment (busbey) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: use substitution

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java [hunk 6]
```diff
@@ -965,11 +964,10 @@ public abstract void recoverFileLease(final FileSystem fs, final Path p,
 
   public static List<Path> getTableDirs(final FileSystem fs, final Path rootdir)
       throws IOException {
-    List<Path> tableDirs = new LinkedList<>();
+    List<Path> tableDirs = new ArrayList<>();
```
Comment (busbey) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: what's your reasoning that ArrayList is a better choice here?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 2 above)
Comment (belugabehr) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: I ran 'blame' on `Path#getName()` and it says the following code has been there for 11 years now:

```
  public String getName() {
    String path = uri.getPath();
    int slash = path.lastIndexOf(SEPARATOR);
    return path.substring(slash+1);
  }
```

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 3 above)
Comment (belugabehr) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: Thanks!

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 4 above)
Comment (belugabehr) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: That method says:

```
@return A map for each table and its percentage.
```

It does not say anything about returning a null value.  I don't think it needs to list everything it won't return.  I think it's appropriate as is.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 5 above)
Comment (belugabehr) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: OK. :)

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 6 above)
Comment (belugabehr) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: Generally speaking ArrayList should be the default choice.

https://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist-in-java

In this case, it makes sense because it there is the advantage of using `addAll` method on the Collection.  The array can be resized very accurately since it's not coming in one-by-one.  Also, if the list returned is large, it is much faster to copy the references into `tableDirs` instead of having to instantiate a LinkedList Node for each value.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 4 above)
Comment (busbey) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: we're a large project with a lot of contributors. If an API does not say that it can't return null and the language spec allows a null to be returned, we have to account for it.

so either

1) change this code to account for a null

or 

2) update the javadoc so the method says it doesn't return null.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 2 above)
Comment (busbey) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: if the api doesn't say it won't return null then we have to account for it possibly being null.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 6 above)
Comment (busbey) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: that sounds reasonable. we're more likely to have a small number of namespaces with lots of tables then have lots of namespaces with very few tables.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 4 above)
Comment (belugabehr) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: I'll go with `update the javadoc so the method says it doesn't return null.`

Thanks.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java (hunk 2 above)
Comment (belugabehr) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: `toSearchName` currently (just a few lines down) does not check for 'null' I will add there too.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 33s |  master passed  |
| +1 :green_heart: |  compile  |   0m 56s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  master passed  |
| +0 :ok: |  spotbugs  |   4m  8s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m  5s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  hbase-server: The patch generated 0 new + 66 unchanged - 8 fixed = 66 total (was 74)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 46s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 20s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 169m 44s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  The patch does not generate ASF License warnings.  |
|  |   | 227m 14s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.quotas.TestClusterScopeQuotaThrottle |
|   | hadoop.hbase.quotas.TestQuotaAdmin |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/912 |
| JIRA Issue | HBASE-23380 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 8deb41272a95 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-912/out/precommit/personality/provided.sh |
| git revision | master / 85a081925b |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/4/testReport/ |
| Max. process+thread count | 5107 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  1s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 26s |  master passed  |
| +1 :green_heart: |  compile  |   0m 56s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 40s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  master passed  |
| +0 :ok: |  spotbugs  |   5m 22s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 19s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 25s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 26s |  hbase-server: The patch generated 0 new + 66 unchanged - 8 fixed = 66 total (was 74)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 41s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 43s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 25s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 157m 30s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 219m 46s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/912 |
| JIRA Issue | HBASE-23380 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4dd2cd85eed6 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-912/out/precommit/personality/provided.sh |
| git revision | master / 85a081925b |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/5/testReport/ |
| Max. process+thread count | 5139 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-912/5/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by belugabehr** [comment]
@busbey Should be good to go here.  Thanks again for the reviews.  Please consider for inclusion into the project.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java [hunk 7]
```diff
@@ -227,12 +232,8 @@ public static FSDataOutputStream create(Configuration conf, FileSystem fs, Path
...
           LOG.debug("DFS Client does not support most favored nodes create; using default create");
-          if (LOG.isTraceEnabled()) LOG.trace("Ignoring; use default create", e);
-        } catch (IllegalArgumentException e) {
-          LOG.debug("Ignoring (most likely Reflection related exception) " + e);
-        } catch (SecurityException e) {
-          LOG.debug("Ignoring (most likely Reflection related exception) " + e);
-        } catch (IllegalAccessException e) {
+          LOG.trace("Ignoring; use default create", e);
+        } catch (IllegalArgumentException | SecurityException |  IllegalAccessException e) {
           LOG.debug("Ignoring (most likely Reflection related exception) " + e);
```
Comment (busbey) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java]: nit: while we're here could we pass this exception rather than coercing it to a string?

**Comment by busbey** [comment]
thanks!
